elkodon_bb/container/src/semantic_string.rs

```diff
@@ -20,6 +22,16 @@ impl std::fmt::Display for SemanticStringError {
 
 impl std::error::Error for SemanticStringError {}
 
+impl From<FixedSizeByteStringModificationError> for SemanticStringError {
+    fn from(value: FixedSizeByteStringModificationError) -> Self {
+        match value {
+            FixedSizeByteStringModificationError::InsertWouldExceedCapacity => {
+                SemanticStringError::ExceedsMaximumLength
+            }
+        }
+    }
+}
```

Comment:
Aaaargh! I realized only now that there is a macro for this. Shall I replace it with the macro?

Reply:
Yes, please use `enum_gen` for this. Then it is at least consistent. Even when we refactor our error handling system we at least find these little things quickly.

Reply:
Done. I'm not that happy though since it now maps the whole `FixedSizeByteStringModificationError` enum to `SemanticStringError::ExceedsMaximumLength`. Is there a way to map only specific enum tags?

Reply:
Sadly no. As soon as elkodon is announced, we have to revisit the error handling concept and make it bright and shiny.

Reply:
Oh, I did not remember my own code well enough anymore :/ ... 
It seems that maybe a generalization would be a better fit, so would this make more sense for you
```rust
enum_gen! {
    SemanticStringError
  generalization:
    ExceedsMaximumLength <= FixedSizeByteStringModificationError,
}
```

Reply:
It is already like that :)